services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java

```diff
@@ -0,0 +1,128 @@
+ * ---license-end
+ */
+
+package app.coronawarn.server.services.distribution.assembly.diagnosiskeys.structure.file;
+
+import app.coronawarn.server.common.persistence.domain.DiagnosisKey;
+import app.coronawarn.server.services.distribution.assembly.structure.WritableOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.directory.Directory;
+import app.coronawarn.server.services.distribution.assembly.structure.directory.DirectoryOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.util.ImmutableStack;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig.Signature;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Set;
+import org.junit.Rule;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.rules.TemporaryFolder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@EnableConfigurationProperties(value = DistributionServiceConfig.class)
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {DistributionServiceConfig.class},
+    initializers = ConfigFileApplicationContextInitializer.class)
+@DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
+class TemporaryExposureKeyExportFileTest {
+
+  public static final String TEK_EXPORT_CHECKSUM_FILE_NAME = "export.bin.checksum";
+
+  @Rule
+  TemporaryFolder outputFolder = new TemporaryFolder();
+
+  @Autowired
+  DistributionServiceConfig distributionServiceConfig;
+
+  @BeforeEach
+  void setup() throws IOException {
+    outputFolder.create();
+  }
+
+  @Test
+  void testChecksumIsDeterministic() throws IOException {
+    File outputFile1 = createFile();
+    byte[] checksum1 = readChecksumFile(outputFile1);
+
+    File outputFile2 = createFile();
+    byte[] checksum2 = readChecksumFile(outputFile2);
+
+    assertThat(Arrays.equals(checksum1, checksum2));
+  }
+
+  @Test
+  void testChecksumChangesWhenPrivateKeyIsModified() throws IOException {
+    File outputFile1 = createFile();
+    byte[] checksum1 = readChecksumFile(outputFile1);
+
+    modifySignaturePrivateKeyVersion();
+
+    File outputFile2 = createFile();
+    byte[] checksum2 = readChecksumFile(outputFile2);
+
+    assertThat(!(Arrays.equals(checksum1, checksum2)));
```

Comment:
Better use of assertions:
```suggestion
    assertThat(checksum1).isNotEqualTo(checksum2);
```

Reply:
Checksum1 and checksum2 are arrays; `isEqualTo` and `isNotEqualTo` is checking for referential equality, which we do not want. We need a check for equal content of the arrays' content.

Reply:
I assume that ``isEqualTo`` is employing ``org.assertj.core.internal.ObjectArrayElementComparisonStrategy`` here, which performs a _propper_ array comparison.
That is also what I could observed when trying it.

Reply:
Okay, actually it uses ``Arrays.deepEquals`` internally.